contracts/integrations/test/internal-integrations-tests/exchange_integration_test.ts

```diff
@@ -0,0 +1,135 @@
+                makerAddress,
+                makerAssetAmount: new BigNumber(1),
+                takerAssetAmount: new BigNumber(1),
+                makerFee: constants.ZERO_AMOUNT,
+                takerFee: constants.ZERO_AMOUNT,
+                feeRecipientAddress,
+            });
+
+            const receipt = await deploymentManager.exchange.fillOrder.awaitTransactionSuccessAsync(
+                order,
+                new BigNumber(1),
+                order.signature,
+                {
+                    from: takers[0],
+                    gasPrice: GAS_PRICE,
+                    value: DeploymentManager.protocolFeeMultiplier.times(GAS_PRICE),
+                },
+            );
+
+            // Ensure that the number of emitted logs is equal to 3. There should have been a fill event
+            // and two transfer events. A 'StakingPoolActivated' event should not be expected because
+            // the only staking pool that was created does not have enough stake.
+            expect(receipt.logs.length).to.be.eq(3);
+
+            // Ensure that the fill event was correct.
+            const fillArgs = filterLogsToArguments<IExchangeFillEventArgs>(receipt.logs, IExchangeEvents.Fill);
+            expect(fillArgs.length).to.be.eq(1);
+            expect(fillArgs).to.be.deep.eq([
+                {
+                    makerAddress,
+                    feeRecipientAddress,
+                    makerAssetData: order.makerAssetData,
+                    takerAssetData: order.takerAssetData,
+                    makerFeeAssetData: order.makerFeeAssetData,
+                    takerFeeAssetData: order.takerFeeAssetData,
+                    orderHash: orderHashUtils.getOrderHashHex(order),
+                    takerAddress: takers[0],
+                    senderAddress: takers[0],
+                    makerAssetFilledAmount: order.makerAssetAmount,
+                    takerAssetFilledAmount: order.takerAssetAmount,
+                    makerFeePaid: constants.ZERO_AMOUNT,
+                    takerFeePaid: constants.ZERO_AMOUNT,
+                    protocolFeePaid: DeploymentManager.protocolFeeMultiplier.times(GAS_PRICE),
+                },
+            ]);
+
+            // Ensure that the transfer events were correctly emitted.
+            const transferArgs = filterLogsToArguments<IERC20TokenTransferEventArgs>(
+                receipt.logs,
+                IERC20TokenEvents.Transfer,
+            );
+            expect(transferArgs.length).to.be.eq(2);
+            expect(transferArgs).to.be.deep.eq([
+                {
+                    _from: takers[0],
+                    _to: makerAddress,
+                    _value: order.takerAssetAmount,
+                },
+                {
+                    _from: makerAddress,
+                    _to: takers[0],
+                    _value: order.makerAssetAmount,
+                },
+            ]);
+        });
```

Comment:
I suspect this is more of a toy example, but we should probably also check that the ETH balance of the staking contract increased when we actually do the Exchange-Staking integration tests.

Reply:
It was intended to be more of a toy example, but yeah I agree that we should add those assertions in the integration tests.